src/composite/tag/tag.jl

```diff
@@ -8,11 +8,14 @@ but tag the block with some extra information.
 """
 abstract type TagBlock{BT, N, T} <: AbstractContainer{BT, N, T} end
 
+# forward content properties
 cache_key(tb::TagBlock) = cache_key(content(tb))
 occupied_locs(x::TagBlock) = occupied_locs(content(x))
 
 Base.:(==)(a::TB, b::TB) where {TB<:TagBlock} = content(a) == content(b)
+Base.length(x::TagBlock) = length(content(x))
```

Comment:
What are the using cases for these forwarded functions?

One example that this may leads to confusion is iterating over`Daggered(chain([a, b, c]))`.

Reply:
It's in test, if you don't forward `length` then collect `MockedTag(chain(X, Y))` won't work. But `collect(chain(X, Y))` work, if this should not work, then we shouldn't forward `iterate` either.

Reply:
finally, we decide to remove this dangerous feature to prevent potential misleading behavior.